feat (Homepage): add OwnProductsSection

yaroslavgorshkov opened the pull request.

yaroslavgorshkov committed:
feat (Homepage): add OwnProductsSection

maxsv01 commented on src/app/components/OwnProductsSection/OwnProductsSectionContent/OwnProductsSectionContent.tsx:
```diff
@@ -0,0 +1,17 @@
+import { OwnProductsSectionContentItemHighlighted } from "./OwnProductsSectionContentItemHighlighted";
+
+export const OwnProductsSectionContent = () => {
+    const ownProductsSectionContent = ownProductsSectionContentMockData;
```
как по мне избыточная переменная и если уже переменная, то у нее должен быть тип 🅰️

yaroslavgorshkov replied:
Исправил

maxsv01 commented on src/app/globals.css:
```diff
@@ -110,4 +110,43 @@ html, body {
+
+.own-products-section-wrapper .own-products-section__content-item:hover .own-products-section__content-item-image-wrapper {
+    filter: blur(5px) contrast(0.8);
+}
```
очень много кастом кода...
в чем была проблема сделать запись через tailwind?

yaroslavgorshkov replied:
Исправил. Такой вопрос. Мы пишем через теилвинд вообще все что возможно написать через теилвинд, а в кастомный цсс пишем только то, что никак через теилвинд не записать, верно?

yaroslavgorshkov committed:
fix (Himepage): minimize css custom code

yaroslavgorshkov committed:
fix (CategorySection, OwnProductsSection): change naming in variables and tailwind classes

yaroslavgorshkov committed:
fix (OwnProductsSection): change imgRef to imgSrc naming in variables and props

yaroslavgorshkov commented on src/app/components/OwnProductsSection/ownProductsSectionMockData.ts:
```diff
@@ -0,0 +1,22 @@
+    {
+        id: 0,
+        href: "#own-bedroom",
+        imgRef: "/img/own-prod-1.jpg",
```
Исправил

yaroslavgorshkov commented on src/app/components/OwnProductsSection/OwnProductsSectionContent/OwnProductsSectionContentItem.tsx:
```diff
@@ -0,0 +1,33 @@
+    }
+
+    return (
+        <article className="own-products-section__content-item group relative w-[231px] h-[324px] rounded-[16px] shadow-lg md:w-[360px] md:h-[507px]">
```
Исправил

yaroslavgorshkov commented on src/app/components/CategorySection/CategorySectionContent/CategorySectionContentItem/CategorySectionContentItem.tsx:
```diff
@@ -31,17 +32,32 @@ export const CategorySectionContentItem = ({
+        twMerge("absolute top-[70%] left-[50%] translate-x-[-50%] translate-y-[-50%] z-10",
+            isHighlightedCategorySectionContentItemButtonClass);
+
+    const isHighlightedCategorySectionContentItemImageClass = isHighlighted ? "blur-[5px] contrast-[0.8]" : "blur-0 contrast-1 transition-all duration-200 group-hover:blur-[5px] group-hover:contrast-[0.8]";
```
Исправил

На счет того, почему 5px. Я попробовал значения из темы самого теилвинда, мне ничего не понравилось оттуда

maxsv01 replied:
используй ближайшее значение, это 4px
в реальной работе у тебя не будет времени думать и примерять. Если не соответствует дизайну, то идешь за дизайнером и тыкаешь его носом, что это не по дизайн системе. Если дизайнера на проекте нету или он из какой то другой конторы или еще что то, то смело берешь ближайшее значение из дизайн системы. По факту, чем меньше времени на таск потратил, тем лучше. Вылизывание досконально до мелочей почти никогда и никому не нужно, нужен хороший работающий и приносящий прибыль продукт

yaroslavgorshkov commented on src/app/components/CategorySection/CategorySectionContent/CategorySectionContentItem/CategorySectionContentItem.tsx:
```diff
@@ -31,17 +32,32 @@ export const CategorySectionContentItem = ({
+        twMerge("absolute left-[50%] translate-x-[-50%] translate-y-[-50%] z-10 whitespace-nowrap",
+            isHighlightedCategorySectionContentItemTextClass);
+
+    const isHighlightedCategorySectionContentItemButtonClass = isHighlighted ? "block" : "hidden group-hover:block";
```
Исправил

maxsv01 commented on src/app/components/CategorySection/CategorySectionMenu/CategorySectionMenuNavigationList/CategorySectionMenuNavigationListItem.tsx:
```diff
@@ -12,8 +12,10 @@ export const CategorySectionMenuNavigationListItem = ({
 }: CategorySectionMenuNavigationListItem) => {
-    const highlightedClass = isHighlighted ? 'navigation-list__item--highlighted' : '';
-    const combinedClassName = twMerge('navigation-list__item', highlightedClass);
+    const highlightedClass = isHighlighted ? 'font-bold after:top-[110%] after:opacity-100' : 'hover:font-bold after:top-[180%] after:opacity-0 after:transition-all after:duration-200 hover:after:opacity-100 hover:after:top-[110%]';
```
вижу, что здесь используется hover...будет ли этот компонент использоваться для мобильной версии? На телефоне ведь нету hover...
🅰️

maxsv01 commented on src/app/components/Header/HeaderDesktop/HeaderDesktopNavigaionList/HeaderDesktopNavigationListItem.tsx:
```diff
@@ -12,8 +12,10 @@ export const HeaderDesktopNavigationListItem = ({
 }: HeaderDesktopNavigationListItemProps) => {
-    const highlightedClass = isHighlighted ? 'navigation-list__item--highlighted' : '';
-    const combinedClassName = twMerge('navigation-list__item', highlightedClass)
+    const highlightedClass = isHighlighted ? 'font-bold after:top-[110%] after:opacity-100' : 'hover:font-bold after:top-[180%] after:opacity-0 after:transition-all after:duration-200 hover:after:opacity-100 hover:after:top-[110%]';
```
а зачем ты повторяешь некоторые значения? например `font-bold`, `after:top-[110%]`. оно есть и там и там? Зачем?

maxsv01 commented on src/app/components/Header/HeaderDesktop/HeaderDesktopNavigaionList/HeaderDesktopNavigationListItem.tsx:
```diff
@@ -12,8 +12,10 @@ export const HeaderDesktopNavigationListItem = ({
-    const combinedClassName = twMerge('navigation-list__item', highlightedClass)
+    const highlightedClass = isHighlighted ? 'font-bold after:top-[110%] after:opacity-100' : 'hover:font-bold after:top-[180%] after:opacity-0 after:transition-all after:duration-200 hover:after:opacity-100 hover:after:top-[110%]';
+    const combinedClassName =
+        twMerge("relative after:content-[''] after:absolute after:left-0 after:bg-lapis after:w-full after:h-[3px]",
```
плис, либо измени шаг в дизайн системе и тогда используй не четные числа, либо используй тему. 
моя рекомендация использовать везде тему. Постарайся свести к минимуму, а то и везде убрать кастом значения, если в этом нету острой необходимости 🅰️

maxsv01 commented on src/app/components/OwnProductsSection/OwnProductsSectionHeadline/OwnProductsSectionHeadline.tsx:
```diff
@@ -0,0 +1,14 @@
+export const OwnProductsSectionHeadline = () => {
+    return (
+        <div className="flex flex-col flex-shrink-0 gap-[32px] w-[195px] px-[16px] py-[50px] rounded-r-[16px] bg-mint
+            md:px-[20px] md:py-[122px] md:w-[400px] xl:pl-[150px] xl:pr-[100px] xl:w-[577px]">
```
`xl:w-[577px]`  ширина будет статической? это правильно?

yaroslavgorshkov committed:
fix (CategorySection, OwnProductsSection):
change fixed width to responsive and custom tailwind css values to tailwind theme values

yaroslavgorshkov committed:
Merge branch 'main' into feat/TASK-4-own-products-section
